### PR TITLE
Create EngineBlock API client and allow fetching of given consents

### PR DESCRIPTION
**Pull request by rjkip**

### CODE REVIEW

**Comment by rjkip** [comment]
:sparkles: @DRvanR @arothuis :sparkles:

File: src/OpenConext/EngineBlockApiClientBundle/Serialisation/ConsentListDeserialiser.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * @param mixed $json
+     * @return Consent
+     *
+     * @SuppressWarnings(PHPMD.UnusedPrivateMethod)
+     */
+    private static function consentFromJson($json)
+    {
+        Assert::keyExists($json, 'service_provider', "Consent JSON structure must contain key 'service_provider'");
```
Comment (arothuis) [inline src/OpenConext/EngineBlockApiClientBundle/Serialisation/ConsentListDeserialiser.php]: Is there a particular reason for using double quotes here?

File: src/OpenConext/Profile/Tests/Value/UrlTest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+{
+    use DataProvider;
+
+    /**
+     * @test
+     * @group value
+     */
+    public function it_can_be_deserialised()
+    {
+        (new Url('https://domain.invalid'));
```
Comment (arothuis) [inline src/OpenConext/Profile/Tests/Value/UrlTest.php]: Is there any way the testing here can be made more explicit?

**Comment by arothuis** [comment]
:+1:

File: app/config/config.yml [hunk 3]
```diff
@@ -69,3 +69,10 @@ surfnet_saml:
             entity_id: %saml_remote_idp_entity_id%
             sso_url: %saml_remote_idp_sso_url%
             certificate: %saml_remote_idp_certificate%
+
+open_conext_engine_block_api_client:
+    http_client:
+        base_url: 'https://engine-api.vm.openconext.org/'
+        verify_ssl: false
```
Comment (DRvanR) [inline app/config/config.yml]: please remove from default config (or set to true as default), safe should be the default :wink:

File: src/OpenConext/EngineBlockApiClientBundle/Serialisation/ConsentListDeserialiser.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+use OpenConext\Profile\Value\Consent\ServiceProvider;
+use OpenConext\Profile\Value\ConsentList;
+use OpenConext\Profile\Value\DisplayName;
+use OpenConext\Profile\Value\EmailAddress;
+use OpenConext\Profile\Value\Entity;
+use OpenConext\Profile\Value\EntityId;
+use OpenConext\Profile\Value\EntityType;
+use OpenConext\Profile\Value\Url;
+
+final class ConsentListDeserialiser
```
Comment (DRvanR) [inline src/OpenConext/EngineBlockApiClientBundle/Serialisation/ConsentListDeserialiser.php]: as discussed, seems to be less a Deserialiser and more a factory.

File: src/OpenConext/Profile/Value/Url.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            'query'    => null,
+            'fragment' => null,
+        ];
+    }
+
+    /**
+     * @param string $scheme
+     * @return bool
+     */
+    public function hasScheme($scheme)
```
Comment (DRvanR) [inline src/OpenConext/Profile/Value/Url.php]: (as discussed) can be removed, so the `parse_url` could go as well?

**Comment by rjkip** [comment]
@DRvanR Do take note of https://github.com/OpenConext/OpenConext-profile/commit/25fa548c4c906a35694dc63f1f973a500541d148

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**